lib/network_ssl/network_ssl.c

```diff
@@ -523,6 +523,11 @@ network_ssl_read_cancel(void * cookie)
 {
 	struct network_ssl_ctx * ssl = cookie;
 
+	/* Cancel the pending immediate event. */
```

Comment:
No, because this would result in the sequence `network_ssl_read`, `network_ssl_write`, `network_ssl_read_cancel` having the write hang.

We should deregister it in `_close` though.

Reply:
right, done.